Gems/PhysX/Code/Tests/EditorTestUtilities.cpp

```diff
@@ -133,6 +134,97 @@ namespace PhysXEditorTests
         return editorEntity;
     }
 
+    EntityPtr CreateBoxColliderEditorEntity(
+        const AZ::Transform& transform,
+        const AZ::Vector3& nonUniformScale,
+        const AZ::Vector3& boxDimensions,
+        const AZ::Vector3& translationOffset,
+        RigidBodyType rigidBodyType)
+    {
+        EntityPtr editorEntity = CreateInactiveEditorEntity("ColliderComponentEditorEntity");
+        auto* colliderComponent = editorEntity->CreateComponent<PhysX::EditorColliderComponent>();
+        editorEntity->CreateComponent<AzToolsFramework::Components::EditorNonUniformScaleComponent>();
+        if (rigidBodyType == RigidBodyType::Dynamic)
+        {
+            editorEntity->CreateComponent<PhysX::EditorRigidBodyComponent>();
+        }
+        else
+        {
+            editorEntity->CreateComponent<PhysX::EditorStaticRigidBodyComponent>();
+        }
+        editorEntity->Activate();
+        AZ::EntityId editorEntityId = editorEntity->GetId();
+
+        AZ::EntityComponentIdPair idPair(editorEntity->GetId(), colliderComponent->GetId());
+        PhysX::EditorPrimitiveColliderComponentRequestBus::Event(
+            idPair, &PhysX::EditorPrimitiveColliderComponentRequests::SetShapeType, Physics::ShapeType::Box);
+
+        AZ::TransformBus::Event(editorEntityId, &AZ::TransformBus::Events::SetWorldTM, transform);
+        AZ::NonUniformScaleRequestBus::Event(editorEntityId, &AZ::NonUniformScaleRequests::SetScale, nonUniformScale);
+        AzToolsFramework::BoxManipulatorRequestBus::Event(idPair, &AzToolsFramework::BoxManipulatorRequests::SetDimensions, boxDimensions);
```

Comment:
That is what I thought too, I found weird that `EditorColliderComponentRequestBus` had functions to handle all primitives' properties except for box, which are handled by `BoxManipulatorRequestBus`. But changing that is out of scope for this change since that means changing how the collider works. I'll make a note of it so we can discuss it later.